src/Microsoft.Repl/Shell.cs

```diff
@@ -17,8 +18,8 @@ public Shell(IShellState shellState)
             ShellState = shellState;
         }
 
-        public Shell(ICommandDispatcher dispatcher, ISuggestionManager suggestionManager = null)
-            : this(new ShellState(dispatcher, suggestionManager))
+        public Shell(ICommandDispatcher dispatcher, ISuggestionManager suggestionManager = null, IConsoleManager consoleManager = null)
+            : this(new ShellState(dispatcher, suggestionManager, consoleManager: consoleManager))
```

Comment:
This is probably a change for a different PR, but should we consider changing this (the ShellState constructor, really) so it requires the constructor parameters and doesn't create defaults if they're null?

Reply:
I think this gets into having a consolidated composition root (really the Shell constructor should not be new'ing ShellState either), and I'm gonna call out-of-scope for this PR.